core-web/libs/sdk/editor/src/lib/sdk-editor.ts

```diff
@@ -0,0 +1,99 @@
+import { CUSTOMER_ACTIONS, postMessageToEditor } from './actions/customer.actions';
+
+//I have two proposals: standalone functions and a object
+//This is the Object, we gain the advantage to have shorts and clearly functions names
+//because the object name is the glibal description
+//like dotSdkCustomerActions.[AN CUSTOMER ACTION]
+export const dotSdkCustomerActions = {
+    setURL(pathname: string) {
+        postMessageToEditor({
+            action: CUSTOMER_ACTIONS.SET_URL,
+            payload: {
+                url: pathname === '/' ? 'index' : pathname?.replace('/', '')
+            }
+        });
+    },
+
+    setBounds(positionData: Object) {
+        postMessageToEditor({
+            action: CUSTOMER_ACTIONS.SET_BOUNDS,
+            payload: positionData
+        });
+    },
+
+    setContentlet(e: PointerEvent) {
+        let target = e.target as HTMLElement;
+
+        if (target.dataset?.['dot'] !== 'contentlet') {
+            target = target.closest('[data-dot="contentlet"]') as HTMLElement;
+        }
+
+        if (!target) {
+            return;
+        }
+
+        const { x, y, width, height } = target.getBoundingClientRect();
+
+        const contentletPayload = JSON.parse(target.dataset?.['content'] ?? '{}');
+
+        postMessageToEditor({
+            action: CUSTOMER_ACTIONS.SET_CONTENTLET,
+            payload: {
+                x,
+                y,
+                width,
+                height,
+                payload: contentletPayload
+            }
+        });
+    },
+
+    eventScrollHandler() {
+        postMessageToEditor({
+            action: CUSTOMER_ACTIONS.IFRAME_SCROLL
+        });
+    },
+
+    checkIsInsideEditor() {
+        postMessageToEditor({
+            action: CUSTOMER_ACTIONS.PING_EDITOR // This is to let the editor know that the page is ready
+        });
+    },
+
+    didContentChange() {
+        return new MutationObserver((mutationsList) => {
+            for (const { addedNodes, removedNodes, type } of mutationsList) {
+                if (type === 'childList') {
+                    const didNodesChanged = [
+                        ...Array.from(addedNodes),
+                        ...Array.from(removedNodes)
+                    ].filter(
+                        (node) => (node as HTMLDivElement).dataset?.['dot'] === 'contentlet'
+                    ).length;
+
+                    if (didNodesChanged) {
+                        postMessageToEditor({
+                            action: CUSTOMER_ACTIONS.CONTENT_CHANGE
+                        });
+                    }
+                }
+            }
+        });
+    }
+};
```

Comment:
Maybe we should use this approach with some difference to make an `init` in order to get the current Host of the dotcms Editor and use it on the post messages as targetOrigin. I think that should be the way to go if we want to catch the target origin and drop the use of `*`. But for me both approaches are right.

Reply:
I don't like any of the two approaches, will be too difficult to explain to the developers how to implement this for their frameworks, I don't have an answer right now, though.

Reply:
Maybe with other names, something more descriptive

I was thinking that we can use other descriptions for the actions

For example
SET_CONTENTLET = notifies that a contentlet has been hovered and the editor should display the options box.

We could rename this to something like

`showEditOptions()
or
notifyShowEditOptions()`

I think this improves the understanding
And maybe a glossary is necessary to this lib

Reply:
Maybe associate this somehow with the items? like `contentlet.pointerEnter` or something like that.

I think we need to go and check other libraries that are facing this same thing.

Reply:
Maybe an object or class separating like
```
export class DotSdkPageEditor{
    editor = {
        updateURL: (url:string) => {},
        didContentChange: () => {},
        eventScrollHandler: () => {},
        isInsideEditor: () => {}
    }

    contentlet = {
        setHoveredContentlet: (e: PointerEvent) => {},
    }
}
```

This is like oher options works, like [Storyblok](https://github.dev/storyblok/storyblok-js-client/blob/3f9fb786a128d4d9ece41ad3c049ec04d2dc6aae/src/index.ts) but the main different is our Editor only manage the Client Actions (for now)

Im looking for separate concerns yet on our editor

Reply:
One think that we also should think about, is that maybe we will need to make more steps to `add` or `remove` events, so maybe we should provide the `connect/disconnect` events. 

I think we should do more research about this and schedule a meeting, for now it feels like throwing rocks to a window that is very high and we are just hoping we break it in some gracious shot.